src/Symfony/Component/Routing/Generator/CompiledUrlGenerator.php

```diff
@@ -62,4 +74,31 @@ public function generate(string $name, array $parameters = [], int $referenceTyp
 
         return $this->doGenerate($variables, $defaults, $requirements, $tokens, $parameters, $name, $referenceType, $hostTokens, $requiredSchemes);
     }
+
+    private function checkDeprecatedCompiledRoutesFormat(array $compiledRoutes): void
+    {
+        if ([] === $compiledRoutes || !isset($compiledRoutes['routes'])) {
```

Comment:
first part of the check is not needed.

```suggestion
        if (!isset($compiledRoutes['routes'])) {
```

Reply:
Adressed.